app/views/assessments/viewGradesheet.html.erb

```diff
@@ -74,28 +75,35 @@
                              else
                                @course.course_user_data
                              end %>
-      <% for c in matching_students.where.not(id: has_submissions) do %>
-          { "id": "",
+        <% url = request.original_url %>
```

Comment:
This style of getting the params does not really follow the rails paradigm, this kind of checking the params can be easily done in the controller action that runs before the view is rendered. Try doing params logic in  the `viewGradesheet` method in `controllers/assessment/grading.rb`

Reply:
actually you wouldn't even need to do anything in the controller, just use `params[:viewall]` which you seem to be already doing for the toggle logic?